### PR TITLE
Default voter registration methods

### PR DESCRIPTION
**Pull request by ericnormand**
Hello!!

[Pivotal Card](https://www.pivotaltracker.com/story/show/103450168)

Would you like to see the wonders of this PR? Well, step right up!!

Did you know that election authorities have *default voter-registration-methods*? You know that whenever they hold an election, that election **inherits the default voter-registration-methods from the authority** that handles registration?

Well, we needed a place to store those registration methods on our authorities, and that's what this does.

We can write election authorities and read them back in. We've got tests for that!

### CODE REVIEW

**Comment by ericnormand** [comment]
There's just one thing I want to do before this is reviewed: I want to upgrade schema.

File: resources/schemas/20151016000-default-voter-reg-methods.edn [hunk 1]
```diff
@@ -0,0 +1,83 @@
+[
+ {:db/id #db/id[:db.part/db]
+  :db/ident :voter-registration-method/type
+  :db/valueType :db.type/keyword
+  :db/cardinality :db.cardinality/one
+  :db/doc "The type of voter-registration-methods (either :in-person, :by-mail, :online)"
```
Comment (danielglauser) [inline resources/schemas/20151016000-default-voter-reg-methods.edn]: This is just the doc so it's probably no big deal but I think there are additional registration options. Here's the list I have:
OVR
NVRF
Election day
State specific reg form
Call clerk

Comment (cap10morgan) [inline resources/schemas/20151016000-default-voter-reg-methods.edn]: OVR -> :online
NVRF -> :by-mail
Election day actually is a different one -> :election-day
State specific reg from -> :by-mail
Call clerk -> we don't support it yet :)

I think we should probably just `s/either/e.g./` in the docstring.

**Comment by ericnormand** [comment]
I talked to Wes about replacing ocd divisions when there's an update (because ocd-division/id is identity). He said it shouldn't, so I'm researching how we can change that.

**Comment by ericnormand** [comment]
Example from http://docs.datomic.com/schema.html#removing-uniqueness-constraint

```
[[:db/retract :person/external-id :db/unique :db.unique/value]
 [:db/add :db.part/db :db.alter/attribute :person/external-id]]
```

**Comment by danielglauser** [comment]
@ericnormand Why would we ever want two different authorities with the same `ocd-id`? I thought the purpose of the `ocd-id` was to be able to tell election authorities apart from one another? Or do I misunderstand the relationship between municipalities and election authorities?

**Comment by cap10morgan** [comment]
@danielglauser Right now that's mainly because ocd-divisions are components of the authorities, and in some places there is both an election authority and a voter registration authority covering the OCD-Division. So there would be 2 ocd-division entities w/ the same id in those cases.

**Comment by danielglauser** [comment]
Given @cap10morgan's response LGTM

File: src/election_authority_works/entities/authority.clj [hunk 2]
```diff
@@ -86,6 +126,31 @@
...
+
+(defn entity->voter-registration-method
+  [{:keys [voter-registration-method/type
+           voter-registration-method/instructions
+           voter-registration-method/url
+           voter-registration-method/supports-iframe
+           voter-registration-method/acceptable-form]}]
+  (db/dissoc-optional-nil-values
+   {:type type
+    :instructions (not-empty (set (map entity->instructions instructions)))
```
Comment (ericnormand) [inline src/election_authority_works/entities/authority.clj]: I assumed we were making a set of these, but it looks like yours is a map of type to text. Can you change that, @cap10morgan ?

Comment (cap10morgan) [inline src/election_authority_works/entities/authority.clj]: The type to text map feels like a better external representation to me. For one example, it's directly usable by notification templates via `{{thingy.instructions.type}}`.

File: src/election_authority_works/entities/authority.clj [hunk 3]
```diff
@@ -80,28 +120,63 @@
    [:street-2]))
 
 (defn entity->instructions
-  [entity]
-  (when (seq (:voting-method/instruction entity))
-    (into {} (for [inst (:voting-method/instruction entity)]
-               [(:instruction/type inst)
-                (:instruction/text inst)]))))
+  [key-ns entity]
+  (let [key (keyword (name key-ns) "instruction")
```
Comment (ericnormand) [inline src/election_authority_works/entities/authority.clj]: Why not just pass in the instructions set?

File: src/election_authority_works/entities/authority.clj [hunk 4]
```diff
@@ -80,28 +120,63 @@
...
+
+(defn entity->voter-registration-method
+  [{:keys [voter-registration-method/type
+           voter-registration-method/instructions
+           voter-registration-method/url
+           voter-registration-method/supports-iframe
+           voter-registration-method/acceptable-form]}]
+  (db/dissoc-optional-nil-values
+   {:type type
+    :instructions (entity->instructions :voter-registration-method instructions)
```
Comment (ericnormand) [inline src/election_authority_works/entities/authority.clj]: Pull the instructions set out here:

```
(entity->instructions instructions)
```

**Comment by ericnormand** [comment]
LGTM

**Pull request merged**